src/NzbDrone.Core/Download/Clients/DownloadStation/DownloadStation.cs

```diff
@@ -0,0 +1,302 @@
+            {
+                var path = data["default_destination"] as string;
+
+                if (Settings.TvCategory.IsNotNullOrWhiteSpace())
+                {
+                    path = string.Format("{0}/.{1}", path, Settings.TvCategory);
+                }            
+
+                return new DownloadClientStatus
+                {
+                    IsLocalhost = Settings.Host == "127.0.0.1" || Settings.Host == "localhost",
+                    OutputRootFolders = new List<OsPath> { _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(path)) }
+                };
+            }
+            catch (DownloadClientException)
+            {
+                _logger.Debug("Failed to get config from Download Station");
+
+                throw;
+            }
+
+            throw new NotImplementedException();
+        }
+
+        public override void RemoveItem(string downloadId, bool deleteData)
+        {
+            _proxy.RemoveItem(downloadId, deleteData, Settings);            
+        }
+
+        protected override string AddFromMagnetLink(RemoteEpisode remoteEpisode, string hash, string magnetLink)
+        {
+            _proxy.AddTorrentFromUrl(magnetLink, GetDownloadDirectory(), Settings);
+
+            var items = _proxy.GetItems(Settings).Where(t => t.Additional.Detail["uri"] == magnetLink);
+
+            if (items.Any())
+            {
+                try
+                {
+                    return items.Single().Id;
+                }
+                catch (InvalidOperationException)
+                {
+                    var ids = string.Join(",", items.Select(t => t.Id));
+                    _logger.Debug(string.Format("Multiple {0} tasks in Download Station: {1}", magnetLink, ids));
+                }
+            }
+            else
+            {
+                _logger.Debug(string.Format("No such task {0} in Download Station", magnetLink));
+            }
+
+            throw new DownloadClientException("Failed to add task to Download Station");
+        }
+
+        private string GetDownloadDirectory()
+        {            
+            if (Settings.TvDirectory.IsNotNullOrWhiteSpace())
+            {
+                return Settings.TvDirectory;
+            }
+            else if (Settings.TvCategory.IsNotNullOrWhiteSpace())
+            {
+                var config = _proxy.GetConfig(Settings);
+                var destDir = (string)config.GetValueOrDefault("default_destination");
+
+                return string.Format("{0}/{1}", destDir.TrimEnd('/'), Settings.TvCategory);
+            }
+
+            return null;
+        }
+
+        protected override string AddFromTorrentFile(RemoteEpisode remoteEpisode, string hash, string filename, byte[] fileContent)
+        {           
+            _proxy.AddTorrentFromData(fileContent, GetDownloadDirectory(), Settings);
+
+            var items = _proxy.GetItems(Settings).Where(t => t.Additional.Detail["uri"] == filename);
```

Comment:
To your first question. No i can't the only way to query the api is using its internal id which you have to get asking for all the files and seek it using the filename. Grrrrr, awfull i wish they would listen...